packages/boot/test/bootstrapTests/ec-contract-upgrade.test.ts

```diff
@@ -0,0 +1,601 @@
+  await Promise.all(
+    smartWallets.map(w =>
+      w.executeOffer({
+        id: PROPOSAL_INV,
+        invitationSpec: {
+          source: 'purse',
+          instance: econCommitteeCharter,
+          description: 'charter member invitation',
+        },
+        proposal: {},
+      }),
+    ),
+  );
+
+  await Promise.all(
+    smartWallets.map(w => {
+      const invitationPurse = w.getCurrentWalletRecord().purses.find(p => {
+        return p.brand.toString().includes('Invitation');
+      });
+      if (!invitationPurse) return null;
+
+      // @ts-ignore
+      const invitation = invitationPurse.balance.value.find(a =>
+        a.description.includes('Voter'),
+      );
+      return w.executeOffer({
+        id: VOTER_INV,
+        invitationSpec: {
+          source: 'purse',
+          instance: economicCommittee,
+          description: invitation.description,
+        },
+        proposal: {},
+      });
+    }),
+  );
+
+  // Proposing a question using first wallet
+  await smartWallets[0].executeOffer({
+    id: getQuestionId(1),
+    invitationSpec: {
+      invitationMakerName: 'VoteOnParamChange',
+      previousOffer: PROPOSAL_INV,
+      source: 'continuing',
+    },
+    offerArgs: {
+      deadline: 1n,
+      instance: VaultFactory,
+      params: { DebtLimit: { brand: debtBrand, value: 100_000_000n } },
+      path: { paramPath: { key: { collateralBrand } } },
+    },
+    proposal: {},
+  });
+  await eventLoopIteration();
+  t.like(smartWallets[0].getLatestUpdateRecord(), {
+    status: { id: getQuestionId(1), numWantsSatisfied: 1 },
+  });
+
+  // Voting on question using first 3 wallets
+  const lastQuestion = unmarshalFromVstorage(
+    storage.data,
+    latestQuestionKey,
+    fromCapData,
+    -1,
+  );
+  await Promise.all(
+    smartWallets.slice(0, 3).map(w =>
+      w.executeOffer({
+        id: getVoteId(1),
+        invitationSpec: {
+          invitationArgs: [
+            [lastQuestion.positions[0]],
+            lastQuestion.questionHandle,
+          ],
+          invitationMakerName: 'makeVoteInvitation',
+          previousOffer: VOTER_INV,
+          source: 'continuing',
+        },
+        proposal: {},
+      }),
+    ),
+  );
+
+  smartWallets.slice(0, 3).forEach(w =>
+    t.like(w.getLatestUpdateRecord(), {
+      status: { id: getVoteId(1), numWantsSatisfied: 1 },
+    }),
+  );
+
+  // Waiting for period to end
+  await advanceTimeTo(2n);
+
+  // Verifying outcome
+  const lastOutcome = unmarshalFromVstorage(
+    storage.data,
+    lastOutcomeKey,
+    fromCapData,
+    -1,
+  );
+
+  t.assert(lastOutcome.outcome === 'win');
+});
+
+test.serial(
+  'check high priority senders before replacing committee',
+  async t => {
+    const { storage } = t.context;
+
+    const data: any = storage.toStorage({
+      method: 'children',
+      args: [highPrioritySenderKey],
+    });
+    t.true(data.length === 6);
+    t.deepEqual(data.sort(), Object.values(wallets).sort());
```

Comment:
The length check seems redundant.

```suggestion
    t.deepEqual(data.sort(), Object.values(wallets).sort());
```